src/devtools/client/webconsole/components/Input/JSTerm.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { Dispatch, SetStateAction, useRef, useState } from "react";
+import actions from "devtools/client/webconsole/actions/index";
+import { useDispatch, useSelector } from "react-redux";
+import { useGetRecording, useGetRecordingId } from "ui/hooks/recordings";
+import { getFrameScope } from "devtools/client/debugger/src/reducers/pause";
+import { getCommandHistory } from "../../selectors/messages";
+import {
+  getAutocompleteMatches,
+  getCursorIndex,
+  insertAutocompleteMatch,
+} from "../../utils/autocomplete";
+import Autocomplete from "./Autocomplete";
+import { UIState } from "ui/state";
+import clamp from "lodash/clamp";
+import { WrappedCodeMirror } from "./CodeMirror";
+
+function useGetMatches(expression: string) {
+  const frameScope = useSelector((state: UIState) => getFrameScope(state, "0:0"));
```

Comment:
Maybe we could improve type of `getFrameScope` with `'${number}:${number}'` for the second param of it.